pennylane/workflow/qnode.py

```diff
@@ -527,7 +527,7 @@ def __init__(
         self.max_expansion = max_expansion
         cache = (max_diff > 1) if cache == "auto" else cache
 
-        if mcm_method not in ("deferred", "one-shot", None):
+        if mcm_method not in ("deferred", "one-shot", "tree-traversal", None):

```

Comment:
Could we rely on the validation in `MCMConfig` so we we don't have to preproduce the checks here?

Just do:
```
    MCMConfig(mcm_method=mcm_method)
```

Might be a bit more extensible.

Reply:
I don't mind either way. I would say it is a tad more related to #5833 .

Reply:
I added this in #5833